### PR TITLE
Fix setValue for TextareaFormField and text based InputFormField

### PR DESCRIPTION
**Pull request by robertfausk**
... especially when there is already a value given for the field.

This PR did not take non-text types and html5 types into account.
For a complete list of types see: https://www.w3schools.com/tags/att_input_type.asp#midcontentadcontainer

### CODE REVIEW

File: src/DomCrawler/Field/FormFieldTrait.php [hunk 1]
```diff
@@ -48,13 +49,17 @@ public function getValue()
...
-        \is_bool($value) ? $this->element->click() : $this->element->sendKeys($value);
+        return $this->element->getAttribute('disabled') ?? false;
     }
 
-    public function isDisabled()
+    private function setTextValue($value): void
     {
-        return $this->element->getAttribute('disabled') ?? false;
+        // Ensure to clean field before sending keys.
+        // Unable use $this->element->clear(); cause it triggers a change event on it's own which is unexpected behavior.
```
[the remark this answers is not in the document]
Comment (dunglas) [reply src/DomCrawler/Field/FormFieldTrait.php]: Could you merge this suggestion please @robertfausk

**Comment by robertfausk** [comment]
Sorry for delay!

**Comment by dunglas** [comment]
Thanks @robertfausk!

File: src/DomCrawler/Field/InputFormField.php [hunk 2]
```diff
@@ -22,6 +22,17 @@ final class InputFormField extends BaseInputFormField
 {
     use FormFieldTrait;
 
+    public function setValue($value)
+    {
+        if (\in_array($this->element->getAttribute('type'), ['text'], true)) {
```
Comment (stof) [inline src/DomCrawler/Field/InputFormField.php]: what about other textual types like `email` or `url` ?

File: tests/fixtures/textarea-form-field.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Form with ChoiceFormField</title>
```
Comment (stof) [inline tests/fixtures/textarea-form-field.html]: Title does not describe them properly

File: src/DomCrawler/Field/InputFormField.php (hunk 2 above)
Comment (robertfausk) [reply src/DomCrawler/Field/InputFormField.php]: > ... 
> This PR did not take non-text types and html5 types into account.
I didn't want to enlarge this PR too much. 
I'll create an issue for more types and create a PR if I find some time.

File: tests/fixtures/textarea-form-field.html (hunk 3 above)
Comment (robertfausk) [reply tests/fixtures/textarea-form-field.html]: You are right. I need to create a new PR because this one is closed, am I right?

Comment (dunglas) [reply tests/fixtures/textarea-form-field.html]: I'll take care of this

Comment (dunglas) [reply tests/fixtures/textarea-form-field.html]: Fixed by 7a041cb.

Comment (stof) [reply tests/fixtures/textarea-form-field.html]: but this PR is *breaking* these types (as they were all considered textual before)

**Pull request merged**